src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -134,12 +134,12 @@ try
     //   the contents of the entire swap chain is redundant, but more importantly because the scroll rect
     //   is the subset of the contents that are being scrolled into. If you scroll the entire viewport
     //   then the scroll rect is empty, which Present1() will loudly complain about.
-    if (_api.invalidatedRows == range<u16>{ 0, _p.s->cellCount.y })
+    if (_p.invalidatedRows == range<u16>{ 0, _p.s->cellCount.y })
     {
         _p.MarkAllAsDirty();
     }
 
-    if (const auto offset = _api.scrollOffset)
+    if (const auto offset = _p.scrollOffset)

```

Comment:
This is a minor bug. It'll not lead to any obvious issues, it's just wrong. `_p.MarkAllAsDirty()` above sets the `_p.scrollOffset` to 0, because it just shouldn't be scrolling anymore when everything's invalid.

I made the rest of the function use `_p` (p as in "rendering Payload"; yes very creative, I know lol) as well to be consistent throughout the function.